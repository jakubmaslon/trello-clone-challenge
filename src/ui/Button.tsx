import styled from "styled-components";

export const Button = styled.button<{ disabled?: boolean }>`
    width: 100%;
    padding: ${props => props.theme.spaces.base};
    border: none;
    background: ${props => props.theme.colors.turquoise};
    color: ${props => props.theme.colors.white};
    text-transform: uppercase;
    cursor: ${props => props.disabled ? "not-allowed" : "pointer"};
    opacity: ${props => props.disabled ? 0.5 : 1};
`;
