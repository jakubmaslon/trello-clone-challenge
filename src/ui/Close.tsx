import styled from "styled-components";

export const Close = styled.div`
    position: absolute;
    top: ${props => props.theme.spaces.base};
    right: ${props => props.theme.spaces.base};
    padding: ${props => props.theme.spaces.base};
    margin: -${props => props.theme.spaces.base};
    cursor: pointer;

    &:before {
        content: "";
        position: absolute;
        top: 50%;
        left: 0;
        right: 0;
        height: 4px;
        background: ${props => props.theme.colors.grey};
        transform: rotate(45deg) scale(0.5);
    }

    &:after {
        content: "";
        position: absolute;
        top: 50%;
        left: 0;
        right: 0;
        height: 4px;
        background: ${props => props.theme.colors.grey};
        transform: rotate(-45deg) scale(0.5);
    }

    &:hover {
        &:before {
        background: ${props => props.theme.colors.darkGrey};
    }

    &:after {
        background: ${props => props.theme.colors.darkGrey};
    }
    }
`;
