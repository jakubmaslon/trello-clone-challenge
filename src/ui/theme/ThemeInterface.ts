interface Colors {
  white: string;
}

interface Spaces {
  base: string;
}

export interface Theme {
  colors: Colors;
  spaces: Spaces;
  name: string;
}
