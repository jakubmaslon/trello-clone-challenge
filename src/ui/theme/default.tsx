import { Theme } from "./ThemeInterface";

export const theme: Theme = {
  colors: {
    white: "#FFFFFF",
    turquoise: "#3C6E71",
    grey: "#D9D9D9",
    blue: "#284B63",
    darkGrey: "#353535",
    lightGrey: "#F2F2F2",
  },
  spaces: {
    base: "1rem",
    half: "0.5rem",
  },
  boxShadows: {
    light: "rgba(0, 0, 0, 0.15) 0px 2px 8px",
    heavy: "rgba(0, 0, 0, 0.56) 0px 22px 70px 4px",
  },
  name: "Default",
};
