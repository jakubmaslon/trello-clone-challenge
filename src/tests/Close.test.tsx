import * as React from "react";

import "@testing-library/jest-dom/extend-expect";
import renderer from 'react-test-renderer';
import { ThemeProvider } from 'styled-components';
import 'jest-styled-components'

import { Close } from '../ui/Close';
import { theme } from "../ui/theme/default";

const withTheme = (element: React.ReactElement) =>
  <ThemeProvider theme={theme}>
    {element}
  </ThemeProvider>

const CloseWTheme = withTheme(<Close />);

describe("Close component tests", () => {
  it('renders correctly', () => {
    const tree = renderer.create(CloseWTheme).toJSON();
    expect(tree).toMatchSnapshot();
  })
})
