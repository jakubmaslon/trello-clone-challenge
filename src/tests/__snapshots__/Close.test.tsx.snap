// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Close component tests renders correctly 1`] = `
.c0 {
  position: absolute;
  top: 1rem;
  right: 1rem;
  padding: 1rem;
  margin: -1rem;
  cursor: pointer;
}

.c0:before {
  content: "";
  position: absolute;
  top: 50%;
  left: 0;
  right: 0;
  height: 4px;
  background: #D9D9D9;
  -webkit-transform: rotate(45deg) scale(0.5);
  -ms-transform: rotate(45deg) scale(0.5);
  transform: rotate(45deg) scale(0.5);
}

.c0:after {
  content: "";
  position: absolute;
  top: 50%;
  left: 0;
  right: 0;
  height: 4px;
  background: #D9D9D9;
  -webkit-transform: rotate(-45deg) scale(0.5);
  -ms-transform: rotate(-45deg) scale(0.5);
  transform: rotate(-45deg) scale(0.5);
}

.c0:hover:before {
  background: #353535;
}

.c0:hover:after {
  background: #353535;
}

<div
  className="c0"
/>
`;
